src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs

```diff
@@ -363,7 +346,7 @@ public async Task StopAsync(CancellationToken cancellationToken = default)
                     {
                         // The task encountered an await; add it to a list to run concurrently.
                         tasks ??= new();
-                        tasks.Add(Task.Run(() => task, token));
+                        tasks.Add(task);
```

Comment:
Reference: https://github.com/dotnet/runtime/issues/93970

Reply:
This `Task.Run` is basically a no-op - the `token` applies only to the running of the lambda, so there's theoretically a way for the cancellation to apply, but it's close to zero and doesn't do what the original author probably thought (it doesn't apply to the returned task).

Reply:
It believe it is a no-op now. I believe it used to support running the sync portion of each task (before the first await) in parallel before.